### DISCUSSION TITLE
Substitue for @NotificationMessage with @SqsListener in 3.0

### DISCUSSION THREAD

**Discussion by mariuszptasinskii** [awspring/spring-cloud-aws#815] [Q&A] [answered]
Hey, first of all, kudos to you all for the work you put in to prepare Spring Cloud AWS for Spring Boot 3!

I have a question regarding the best replacement for @NotificationMessage that was used with @SqsListener in previous versions:

```
@SqsListener("\${sqs.queues.queue-name}", deletionPolicy = ON_SUCCESS)
    fun consumeMessage(@NotificationMessage event: String) {
        logic()
    }
```
That allowed us to work with just a message payload part of the `Message`, while migrating to 3.0 I've noticed that there's no @NotificationMessage, and in listeners, we receive the whole message:
```
{
  "Type": "Notification",
  "MessageId": "51abff97-11d1-4881-b7bf-fc894645e043",
  "TopicArn": "topic-arn",
  "Message": "message"
  "Timestamp": "2023-05-12T11:21:23.529Z",
  "SignatureVersion": "1",
  "Signature": "xxx",
  "SigningCertURL": "xxx"
}
```
Since we're not really interested in all this data in our listeners I was wondering what's the best way to get just the message payload? Based on [this discussion](https://github.com/awspring/spring-cloud-aws/discussions/760) I was able to make it work with something like this, but I wonder if this is now the best way to make it work.
```
@Bean
    fun defaultSqsListenerContainerFactory(
        sqsAsyncClient: SqsAsyncClient,
        objectMapper: ObjectMapper
    ): SqsMessageListenerContainerFactory<Any> {
        return SqsMessageListenerContainerFactory
            .builder<Any>()
            .configure { options ->
                options.messageConverter(createdNestedMessageConverter(objectMapper))
            }
            .sqsAsyncClient(sqsAsyncClient)
            .build()
    }

    private fun createdNestedMessageConverter(om: ObjectMapper): SqsMessagingMessageConverter {
        return object : SqsMessagingMessageConverter() {
            override fun getPayloadToDeserialize(message: Message): Any {
                return om.readTree(message.body()).get("Message").textValue()
            }
        }
    }
```

**Comment by Stef569** [comment, marked as answer]
Simply create a pojo with 1 field Message as parameter
```
public void getEventFromQueue( NotificationEvent event ) {
event.getMessage();
}
```

**Reply by mariuszptasinskii** [reply]
You're right, that would do it. However I've decided to stick with message converter as it allows me convert message to data class and since not all my listeners expect the same message I can use dedicated container factory for it.
